libraries/core_libs/consensus/src/dag/dag_manager.cpp

```diff
@@ -282,10 +282,11 @@ std::vector<blk_hash_t> DagManager::getDagBlockOrder(blk_hash_t const &anchor, P
   return blk_orders;
 }
 
-void DagManager::clearLightNodeHistory(bool force) {
-  // Actual history size will be between 100% and 110% of light_node_history_ to avoid deleting on every period
-  if (((period_ % (std::max(light_node_history_ / 10, (uint64_t)1)) == 0) || force) && period_ > light_node_history_ &&
-      dag_expiry_level_ > max_levels_per_period_ + 1) {
+void DagManager::clearLightNodeHistory(bool initial) {
+  // Delete once size 1% over the light_node_history_, by default light node history is a week of data so this condition
+  // will pass once an hour
+  if (((period_ % (std::max(light_node_history_ / 100, (uint64_t)1)) == 0) || initial) &&
```

Comment:
Maybe it will be good to move parts of this checks to separate variables to increase readability